### ISSUE TITLE
Red Screen during streaming video from any site 

### ISSUE THREAD

**Issue by Civa69** [mayankk2308/purge-wrangler#30]
### System Configuration
Describe your system configuration here. State any anomalies (USB drives, etc.) involved as well.
- **Mac**: Mac Mini Late 2012, quad core i7, 16 GB RAM, twin SSDs 
- **macOS Version**: 10.13.4, 10.14.5, etc.
- **System GPUs**: Intel HD Graphics 4000
- **eGPU**: Akitio Node Pro, MSI Vega 56 OC Air Boost
- **Script Version**: Unsure. Most Recent, from this GitHub
### Issue Description
I get a red screen every time I attempt to stream video. Nearly everything else works except streaming video from the internet. 

### Expected Behavior
Clear streaming video

### More Context
I have done a system diagnosis, and this was the result: 

Detected System GPUs:
Intel HD Graphics 4000
Radeon RX Vega 56

No anomalies expected.

My system status gives this result: 
>> System Status

Ti82 Devices      Disabled
TB1/2 AMD eGPUs   Enabled
Legacy AMD eGPUs  Enabled
NVIDIA eGPUs      Disabled
Web Drivers       Absent

As I said, everything works except a red screen for ANY streaming video from ANY site; Amazon, hulu, Netflix. All give a red screen during streaming video. Sound comes in fine, just a red screen for video. 

Please advise if there is anything else about my system configuration you need. 

Thank you.

**Comment by mayankk2308** [comment]
Thank you for the report. Legacy GPUs need not be enabled for this GPU. Please uninstall and re-setup.

Do you see the issue in all web browsers?

If after re-setting up, you still see the issue, them this is not an issue related to the script or the patch. To eliminate all variables, it would be best to test on a clean installation. You can do so by creating a new APFS volume via Disk Utility and installing macOS on it. You mention nearly everything else working, so I presume the GPU worked fine in stress tests or benchmarks, as otherwise a likely cause could be faulty hardware.

**Comment by Civa69** [comment]
Please excuse my ignorance, however, how do I stress test? 

This is all new to me, and I’m just finally glad to have two 34 inch ultrawide monitors finally working, albeit the web streaming video issue. 

I also tested with my Plex server, and that video streams fine, over WiFi, to my Mac (it’s not directly connected) 

So I can only surmise it’s only affecting web streaming, for some reason. 

Since Catalina has come out, I can wipe backup the system SSD, wipe it, and do a clean install if Catalina, though that will have to be after work, today (is the 1:30 am, here), however I will also uninstall purge wrangler and reinstall per you suggestion, first. 

I can also stress test, if you can tell me how. 

I should also add both monitors are connected directly to the eGPU, via display port, and display port 1.2 is enabled. I also did a test disabling it, with no change. 

Thank you for your prompt reply. 

Regards, 
– Çiva

**Comment by mayankk2308** [comment]
Run a benchmark such as Unigine Heaven for awhile (15-30m) to stress test GPU. GPU should not crash.

You can use this method to install another copy of macOS without deleting your current system: 
https://www.imore.com/how-to-partition-your-mac

I’m in the same timezone.

**Comment by Civa69** [comment]
Apologies.

I did not realize you were in the same time zone. 

I am running Unigine, and have amped up all the settings to extreme. 

It’s a little blurry, As though it expects me to be wearing 3D glasses, but seems to be running fine. It has about 15 more minutes to run. 

I should also add I checked another site, and the video on there ran fine, so it’s basically Hulu, Netflix, amazon, and I believe DC Universe, that give me the red screen when playing video. 

In other words, all the channels I’m subscribed to, and watch. 😆😆😆😅

**Comment by Civa69** [comment]
It seemed to benchmark fine. I have dropped the PDF of the results in this reply. I am not sure if the PDF works for you, but I have dropped it anyway. 

[Screen Shot 2019-10-08 at 2.41.56 AM.pdf](https://github.com/mayankk2308/purge-wrangler/files/3701064/Screen.Shot.2019-10-08.at.2.41.56.AM.pdf)
[Screen Shot 2019-10-04 at 10.06.25 PM.pdf](https://github.com/mayankk2308/purge-wrangler/files/3701065/Screen.Shot.2019-10-04.at.10.06.25.PM.pdf)
[Screen Shot 2019-10-04 at 10.06.08 PM.pdf](https://github.com/mayankk2308/purge-wrangler/files/3701066/Screen.Shot.2019-10-04.at.10.06.08.PM.pdf)
[Screen Shot 2019-10-04 at 10.05.02 PM.pdf](https://github.com/mayankk2308/purge-wrangler/files/3701067/Screen.Shot.2019-10-04.at.10.05.02.PM.pdf)
[Screen Shot 2019-10-04 at 10.04.29 PM.pdf](https://github.com/mayankk2308/purge-wrangler/files/3701068/Screen.Shot.2019-10-04.at.10.04.29.PM.pdf)
[Screen Shot 2019-10-04 at 10.03.54 PM.pdf](https://github.com/mayankk2308/purge-wrangler/files/3701069/Screen.Shot.2019-10-04.at.10.03.54.PM.pdf)
[Screen Shot 2019-10-04 at 10.03.11 PM.pdf](https://github.com/mayankk2308/purge-wrangler/files/3701070/Screen.Shot.2019-10-04.at.10.03.11.PM.pdf)
[Screen Shot 2019-10-04 at 10.02.23 PM.pdf](https://github.com/mayankk2308/purge-wrangler/files/3701071/Screen.Shot.2019-10-04.at.10.02.23.PM.pdf)
[resumewmata2018.pdf](https://github.com/mayankk2308/purge-wrangler/files/3701072/resumewmata2018.pdf)

**Comment by Civa69** [comment]
Okay, I did the benchmark and everything was okay. I ran it for 45 minutes. I put the benchmark results above. I also uninstalled, tested, streaming video worked fine, reinstalled, tested, still a red screen on streaming video from Netflix, Hulu, Amazon, and DC Universe. 

For kicks, I checked an adult site, and streaming worked fine. 

The only thing I have left is a clean install of Mac OS, which I will do after work, today, which I am late getting to.

**Comment by mayankk2308** [comment]
What web browser are you using?

**Comment by Civa69** [comment]
Safari 

I only use safari, these days 

I also spoke with my IT guy, and he thinks it may be a DRM issue that needs to be activated in the GPU. 

Is this true? If true, then how would I do that? The application for the MSI Vega 56 is windows.

**Comment by mayankk2308** [comment]
By default, DRM content works fine on RX 570 or better - so this should not be an issue. Trying clean install first will give more insight. If it doesn't work there, you should return/replace your GPU.

Since this is not an issue with the script or patches, I'm marking the problem as "closed". However keep me posted here so we can resolve this.

**Comment by Civa69** [comment]
I did more research, and spoke with more IT people 
DRM is a driver issue, not a card issue 

I will do a clean install this weekend, to see if Mac OS dropped the drivers, somehow, and am doing additional testing to see if there may be a bandwidth issue with thunderbolt one (my system is a late 2012 Mac Mini, so, thunderbolt one), and it has also been suggested I attempt to test gradient changes in resolution and color filtering (not sure if I can do that, but will check the color options in the settings function), as those are the only things I can do, with AMD not making drivers for Mac that are downloadable.

However, it is absolutely not a card issue, and fully a software issue.

**Comment by mayankk2308** [comment]
> I did more research, and spoke with more IT people DRM is a driver issue, not a card issue

Except if the card has bad firmware, which can cause issues with the drivers which in turn causes the DRM issue. XFX is a good example - GPUs that are supposed to work natively don't work sometimes.

I've been suggesting a clean install since my very first response for this issue to eliminate all variables. Hopefully that resolves the issue. Keep me posted!

**Comment by Civa69** [comment]
The firmware (bios) can be updated though, correct? 
One of my next steps is to take the card to my office Sys Admin (he and I already discussed this) to have him reflash the bios of the Vega 56 with the most updated bios, tomorrow. 

I still plan to do a clean install, because it’s time to do one anyway, however if the issue is firmware (potentially), that can be corrected by updating the card bios on a Windows system, correct? 

And, of course, I will keep you updated.

**Comment by mayankk2308** [comment]
Yes, a BIOS reflash is a good decision, particularly if the card is used. It can be done on bootcamp too. Again, the easiest solution is to check with an extra clean install as suggested above.

**Comment by Civa69** [comment]
Good news and bad news 

The good news is that it was definitely a software issue. It was the OS, which somehow became corrupted. 

Also, good news, it works in MacOS Catalina 

The BAD NEWS is the Mac will not see my time machine on my NAS, which is a QNAP, saying the server is unavailable, even though I can login to the server. 

It said it was unavailable during install, and it’s unavailable during migration assistant, even though it sees it, and knows it exists. 

So I have a nice fresh clean install, eGPU works, can play media from DRM sites, but can’t access all of my past data. 

Wonderful.

**Comment by Civa69** [comment]
Good news, it works in MacOS Catalina 

BAD NEWS, the Mac will not see my time machine on my NAS, saying it’s unavailable 

WORSE NEWS! 

I think I found the problem, and it is FireWire 800 
Everything worked perfectly, and then I installed the software for my FireWire audio interface 

Red screen all over again 

So the only thing that changed between no red screen, and red screen, was adding a piece of software that allows me to use my FireWire for an audio interface. 

Are we sure this is not a conflict between purge-wrangler, thunderbolt, and FireWire?

**Comment by Civa69** [comment]
Apologies, I checked again to confirm. I plugged one of the displays in as hdmi, and video WILL stream from DRM sites, via the HDMI monitor. 

When I first tested, it was with no HDMI connected, and worked 

I then installed the application for my audio interface, and the red screen returns, when connected to the GPU, over DisplayPort 

One monitor is now connected via display port, the other HDMI 

I even tested by moving the video window to the display port monitor 

Red Screen 

Back to the HDMI monitor, video is clear 

I don’t know if it’s a resolution issue (which didn’t make sense, considering it worked BEFORE I installed the FireWire interface), or if it’s a conflict with FireWire, I can only say FireWire seems to be the issue.

**Comment by mayankk2308** [comment]
> Are we sure this is not a conflict between purge-wrangler, thunderbolt, and FireWire?

Yes. The script only enables GPUs on older Thunderbolt interfaces. This does not affect any Thunderbolt devices or capability. All graphics drivers and anything else is untouched. You can investigate what exactly the script changes by going through it’s source code - which is the primary reason why I’ve kept it open source. 

> I then installed the application for my audio interface, and the red screen returns, when connected to the GPU, over DisplayPort.

Clearly the software you are installing has something to do with it. If the issue does not occur on a clean install, the script has done its job.

**Comment by Civa69** [comment]
I have to get Ethernet cable tomorrow, which will enable me to do another clean install (my time machine backups are on a server, and I apparently have to have a direct connection, which is difficult, due to the distance from my modem to the server, hence the need for 50 feet of cable) 

I will retest once that occur, by not installing the software, first. 

I will update you later today on the results.
